src/pint/toa.py

```diff
@@ -1400,6 +1408,8 @@ def compute_posvels(self, ephem=None, planets=False):
             else:
                 ssb_obs = site.posvel(tdb, ephem)
 
+            self.ssb_obs = ssb_obs
```

Comment:
The test failure is because this line isn't safe. It is saving the `ssb_obs` vector for each of the observatory groups (so after the loop it will be holding the `ssb_obs` for the *last* observatory), then it is getting used later (L1454) for the *first* observatory.  Rather than caching `ssb_obs` in `self.ssb_obs`, it would be better to calculate it on the fly with `site.posvel()`